editor/src/clj/editor/editor_extensions/ui_docs.clj

```diff
@@ -0,0 +1,718 @@
+  (component
+    "paragraph"
+    :description "Typography component representing a paragraph of text"
+    :props paragraph-props))
+
+(def heading-component
+  (component
+    "heading"
+    :description "Typography component representing a heading"
+    :props heading-props))
+
+(def icon-component
+  (component
+    "icon"
+    :description "An icon of a fixed size"
+    :props icon-props))
+
+(def button-component
+  (component
+    "button"
+    :props button-props
+    :description "Button with a label and/or an icon"))
+
+(def check-box-component
+  (component
+    "check_box"
+    :props check-box-props
+    :description "Check box with a label"))
+
+(def select-box-component
+  (component
+    "select_box"
+    :props select-box-props
+    :description "Dropdown select box with an array of options"))
+
+(def text-field-component
+  (component
+    "text_field"
+    :props text-field-props
+    :description "Single-line text field, reports changes on typing"))
+
+(def value-field-component
+  (component
+    "value_field"
+    :props generic-value-field-props
+    :description "Input component based on a text field, reports changes on commit (<code>Enter</code> or focus loss).\n\nSee also: <code>string_field</code>, <code>number_field</code> and <code>integer_field</code>"))
+
+(def string-field-component
+  (component
+    "string_field"
+    :props custom-value-field-props
+    :description "String input component based on a text field, reports changes on commit (<code>Enter</code> or focus loss)"))
+
+(def integer-field-component
+  (component
+    "integer_field"
+    :props custom-value-field-props
+    :description "Integer input component based on a text field, reports changes on commit (<code>Enter</code> or focus loss)"))
+
+(def number-field-component
+  (component
+    "number_field"
+    :props custom-value-field-props
+    :description "Number input component based on a text field, reports changes on commit (<code>Enter</code> or focus loss)"))
+
+(def dialog-button-component
+  (component
+    "dialog_button"
+    :props dialog-button-props
+    :description "Dialog button shown in the footer of a dialog"))
+
+(def dialog-component
+  (component
+    "dialog"
+    :props dialog-props
+    :description "Dialog component, a special component that can't be used as a child of other components"))
+
+;; endregion
+
+;; region docs
+
+(defn- component->script-doc [{:keys [name props description]}]
+  (let [{req true opt false} (group-by :required props)]
+    {:name name
+     :type :function
+     :description description
+     :parameters [{:name "props"
+                   :types ["table"]
+                   :doc (str (when req
+                               (str "Required props:\n"
+                                    (props-doc-html req)
+                                    "\n\n"))
+                             "Optional props:\n"
+                             (props-doc-html opt))}]
+     :returnvalues [{:name "value"
```

Comment:
Yes, we did, from `sdoc` files. We also use these maps to write the editor's script doc files, so it's just more convenient to use this keyword despite the readability issues. Otherwise, we would need to convert it back and forth.